packages/@o3r/localization/builders/helpers/localization.generator.ts

```diff
@@ -362,7 +375,15 @@ export class LocalizationExtractor {
           .forEach((locKey) => addMetadata(this.generateMetadataItemFromLocalization(localizationMap[locFile].data[locKey], locKey), locFile))
       );
 
-    const localizationMetadata = Object.keys(metadata).map((k) => metadata[k]);
+    const localizationMetadata = Object.values(metadata);
+
+    localizationMetadata.forEach((data) => {
+      if (data.ref && !data.ref.includes(referenceSeparator) && !metadata[data.ref]) {
+        this.logger.error(`The key ${data.ref} is unknown but referenced by ${data.key}.`);
+        throw new O3rCliError('Unknown referenced key');
```

Comment:
Do you want to throw at the first key of we way to report them all before throwing?